Classes/Domain/Resource/GenericModelResource.php

```diff
@@ -56,6 +58,9 @@ public function initializeObject()
         $this->relationshipsToBeApiExposed = $settings['relationshipsToBeApiExposed'];
         ksort($this->relationshipsToBeApiExposed);
         $this->identityAttributes = $settings['identityAttributes'];
+        if ($this->payload instanceof ExposedValueObjectInterface) {
```

Comment:
We could make this an `AfterAspect` to `ExposedValueObjectInterface::initializeObject()` to have it completely out of the JsonApi creation and lookup mechanism and have it directly connected to the value object itself.